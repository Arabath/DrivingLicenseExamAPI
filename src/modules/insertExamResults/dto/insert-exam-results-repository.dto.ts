export class InsertExamResultsRepositoryDto {
    idTurnoAsignado: number;
    estado: number;
    tipo: string;
    constructor({
      idTurnoAsignado,
      estado,
      tipo,
    }: {
      idTurnoAsignado: number;
      estado: number;
      tipo: string;
    }) {
      this.idTurnoAsignado = idTurnoAsignado;
      this.estado = estado
      this.tipo = tipo;

    }
  }
  