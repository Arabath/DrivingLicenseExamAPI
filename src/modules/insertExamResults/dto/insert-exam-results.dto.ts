import { ApiProperty } from '@nestjs/swagger';

export class InsertExamResultsDto {
  @ApiProperty()
  idTurnoAsignado: number;
  @ApiProperty()
  estado: number;
  @ApiProperty()
  tipo: string;
}
