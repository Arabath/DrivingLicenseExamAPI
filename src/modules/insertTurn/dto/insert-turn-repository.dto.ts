export class InsertTurnRepositoryDto {
  idTurno: number;
  idSolicitudTurno?: number;
  tipoDocumento: string;
  documento: string;
  apellido: string;
  nombre: string;
  idPais?: number;
  telefono: string;
  fechaNacimiento: Date;
  domicilio: string;
  domicilioNumero: String;
  entreCalle1?: string;
  entreCalle2?: string; 
  codigoPostal: string;
  localidad: string;
  idTramite: number;
  fechaRenovacion?: Date;
  email: string;
  token: string;
  grupoFactor?: string;
  constructor({
    idTurno,
    idSolicitudTurno,
    tipoDocumento,
    documento,
    apellido,
    nombre,
    idPais,
    fechaNacimiento,
    telefono,
    domicilio,
    domicilioNumero,
    entreCalle1,
    entreCalle2,
    codigoPostal,
    localidad,
    idTramite,
    fechaRenovacion,
    email,
    token,
    grupoFactor,
  }: {
    idTurno: number;
    idSolicitudTurno?: number;
    tipoDocumento: string;
    documento: string;
    apellido: string;
    nombre: string;
    idPais?: number;
    fechaNacimiento: Date;
    telefono: string;
    domicilio: string;
    domicilioNumero: String;
    entreCalle1?: string;
    entreCalle2?: string;
    codigoPostal: string;
    localidad: string;
    idTramite: number;
    fechaRenovacion?: Date;
    email: string;
    token: string;
    grupoFactor?: string;
  }) {
    this.idTurno = idTurno;
    this.idSolicitudTurno = idSolicitudTurno
    this.tipoDocumento = 'CUIL';
    this.documento = documento;
    this.apellido = apellido;
    this.nombre = nombre;
    this.idPais = 11;
    this.fechaNacimiento = fechaNacimiento;
    this.telefono = telefono;
    this.domicilio = domicilio;
    this.domicilioNumero = domicilioNumero;
    this.entreCalle1 = entreCalle1;
    this.entreCalle2 = entreCalle2;
    this.codigoPostal = '2804';
    this.localidad = 'Campana';
    this.idTramite = idTramite;
    this.fechaRenovacion = fechaRenovacion;
    this.email = email;
    this.token = token;
    this.grupoFactor = grupoFactor;
  }
}
