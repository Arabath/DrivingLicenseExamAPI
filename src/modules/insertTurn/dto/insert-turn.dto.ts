import { ApiProperty } from '@nestjs/swagger';

export class InsertAssignedTurnDto {
  @ApiProperty()
  idTurno: number;
  @ApiProperty()
  idSolicitudTurno?: number;
  @ApiProperty()
  tipoDocumento: string;
  @ApiProperty()
  documento: string;
  @ApiProperty()
  apellido: string;
  @ApiProperty()
  nombre: string;
  @ApiProperty()
  idPais?: number;
  @ApiProperty()
  telefono: string;
  @ApiProperty()
  fechaNacimiento: Date;
  @ApiProperty()
  domicilio: string;
  @ApiProperty()
  domicilioNumero: String;
  @ApiProperty()
  entreCalle1?: string;
  @ApiProperty()
  entreCalle2?: string;
  @ApiProperty()
  codigoPostal: string;
  @ApiProperty()
  localidad: string;
  @ApiProperty()
  idTramite: number;
  @ApiProperty()
  fechaRenovacion?: Date;
  @ApiProperty()
  email: string;
  @ApiProperty()
  token: string;
  @ApiProperty()
  grupoFactor?: string;
}
