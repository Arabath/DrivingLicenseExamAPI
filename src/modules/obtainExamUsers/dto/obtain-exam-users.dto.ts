import { ApiProperty } from '@nestjs/swagger';

export class UsersExamDto {
  @ApiProperty()
  EstadoExamen: number;
  @ApiProperty()
  FechaExamen: Date;
  constructor({
    EstadoExamen,
    FechaExamen,
  }: {
    EstadoExamen: number;
    FechaExamen: Date;
  }) {
    (this.EstadoExamen = EstadoExamen), (this.FechaExamen = FechaExamen);
  }
}