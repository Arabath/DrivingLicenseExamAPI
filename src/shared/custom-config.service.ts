import { Inject, Injectable } from '@nestjs/common';
import { TypeOrmModuleOptions } from '@nestjs/typeorm';
import { ConfigService } from '@nestjs/config';
//import { DataSourceOptions } from 'typeorm';

@Injectable()
export class CustomConfigService {
  constructor(@Inject(ConfigService) private configService: ConfigService) {}

  get databaseConfig(): TypeOrmModuleOptions {
    if (this.configService.get('NODE_ENV') == 'test') {
      return {
        type: 'mssql',
        host: this.configService.get('DB_HOST'),
        port: parseInt(this.configService.get('DB_PORT')),
        username: this.configService.get('DB_USERNAME'),
        password: this.configService.get('DB_PASSWORD'),
        database: this.configService.get('DB_DATABASE_TESTING'),
        entities: [],
        synchronize: false,
        options: { encrypt: false },
      };
    } else {
      return {
        type: 'mssql',
        host: this.configService.get('DB_HOST'),
        port: parseInt(this.configService.get('DB_PORT')),
        username: this.configService.get('DB_USERNAME'),
        password: this.configService.get('DB_PASSWORD'),
        database: this.configService.get('DB_DATABASE'),
        entities: [],
        synchronize: false,
        options: { encrypt: false },
      };
    }
  }

  get mailerConfig() {
    return {
      host: this.configService.get('MAIL_HOST'),
      port: parseInt(this.configService.get('MAIL_PORT')),
      secure: true,
      auth: {
        user: this.configService.get('MAIL_USER'),
        pass: this.configService.get('MAIL_PASSWORD'),
      },
      tls: {
        ciphers: 'SSLv3',
      },
    };
  }
}
