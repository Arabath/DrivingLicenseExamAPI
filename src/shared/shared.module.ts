import { Global, Module } from '@nestjs/common';
import { CustomConfigService } from './custom-config.service';
//import { ConfigModule, ConfigService } from '@nestjs/config';

@Global()
@Module({
  providers: [CustomConfigService],
  exports: [CustomConfigService],
})
export class SharedModule {}
