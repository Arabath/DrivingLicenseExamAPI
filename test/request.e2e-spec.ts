import { INestApplication } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';
import { CustomConfigService } from '../src/shared/custom-config.service';
import { Test } from '@nestjs/testing';
import { ConfigModule } from '@nestjs/config';
import { RequestRepository } from '../src/modules/request/request.repository';
import { SharedModule } from '../src/shared/shared.module';
import { RequestModule } from '../src/modules/request/request.module';
import { CreateRequestRepositoryDto } from '../src/modules/request/dto/create-request-repository.dto';
import * as request from 'supertest';
import { RequestResponseDto } from '../src/modules/request/dto/request-response.dto';
import e from 'express';

describe('Request', () => {
  let app: INestApplication;
  let requestRepository: RequestRepository;

  const testRequest = {
    nombre: 'Mariano',
    apellido: 'Baldovino',
    email: 'mariano1baldovino@gmail.con',
    documento: '45051325',
    idTramite: 1,
    token: 'tokenfalso123',
  };

  const createRequestRepositoryDto = new CreateRequestRepositoryDto({
    ...testRequest,
  });

  beforeEach(async () => {
    const moduleRef = await Test.createTestingModule({
      imports: [
        ConfigModule.forRoot({ isGlobal: true }),
        SharedModule,
        TypeOrmModule.forRootAsync({
          inject: [CustomConfigService],
          useFactory: async (cf: CustomConfigService) => cf.databaseConfig,
        }),
        RequestModule,
      ],
      providers: [CustomConfigService],
    }).compile();

    app = moduleRef.createNestApplication();
    await app.init();

    requestRepository = moduleRef.get<RequestRepository>(RequestRepository);
    requestRepository.deleteAll()

  });

  it('/GET request', async () => {
    const res = await requestRepository.create(createRequestRepositoryDto);

    return request(app.getHttpServer())
      .get('/request/' + testRequest.token)
      .expect(200)
      .then((response) => {
        const data = response.body.data
        delete data.id
        expect(data).toMatchSnapshot()
      });
  });

  afterAll(async () => {
    await app.close();
  });
});
