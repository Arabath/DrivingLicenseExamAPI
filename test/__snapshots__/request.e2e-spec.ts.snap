// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request /GET request 1`] = `
{
  "apellido": "Baldovino",
  "codigoTramite": null,
  "documento": "45051325",
  "email": "mariano1baldovino@gmail.con",
  "idTramite": 1,
  "nombre": "Mariano",
}
`;
